nifi-api/src/main/java/org/apache/nifi/time/TimeFormat.java

```diff
@@ -0,0 +1,251 @@
+            return unitMultipliers.stream().reduce(1L, (a, b) -> (long) a * b);
+        }
+    }
+
+    /**
+     * Returns the next smallest {@link TimeUnit} (i.e. {@code TimeUnit.DAYS -> TimeUnit.HOURS}).
+     * If the parameter is {@code null} or {@code TimeUnit.NANOSECONDS}, an
+     * {@link IllegalArgumentException} is thrown because there is no valid smaller TimeUnit.
+     *
+     * @param originalUnit the TimeUnit
+     * @return the next smaller TimeUnit
+     */
+    TimeUnit getSmallerTimeUnit(TimeUnit originalUnit) {
+        if (originalUnit == null || TimeUnit.NANOSECONDS == originalUnit) {
+            throw new IllegalArgumentException("Cannot determine a smaller time unit than '" + originalUnit + "'");
+        } else {
+            return TimeUnit.values()[originalUnit.ordinal() - 1];
+        }
+    }
+
+    /**
+     * Returns {@code true} if this raw unit {@code String} is parsed as representing "weeks", which does not have a value in the {@link TimeUnit} enum.
+     *
+     * @param rawUnit the String containing the desired unit
+     * @return true if the unit is "weeks"; false otherwise
+     */
+    private boolean isWeek(final String rawUnit) {
+        return switch (rawUnit) {
+            case "w", "wk", "wks", "week", "weeks" -> true;
+            default -> false;
+        };
+    }
+
+    /**
+     * Returns the {@link TimeUnit} enum that maps to the provided raw {@code String} input. The
+     * highest time unit is {@code TimeUnit.DAYS}. Any input that cannot be parsed will result in
+     * an {@link IllegalArgumentException}.
+     *
+     * @param rawUnit the String to parse
+     * @return the TimeUnit
+     */
+    protected TimeUnit determineTimeUnit(String rawUnit) {
+        return switch (rawUnit.toLowerCase()) {
+            case "ns", "nano", "nanos", "nanoseconds" -> TimeUnit.NANOSECONDS;
+            case "µs", "micro", "micros", "microseconds" -> TimeUnit.MICROSECONDS;
+            case "ms", "milli", "millis", "milliseconds" -> TimeUnit.MILLISECONDS;
+            case "s", "sec", "secs", "second", "seconds" -> TimeUnit.SECONDS;
+            case "m", "min", "mins", "minute", "minutes" -> TimeUnit.MINUTES;
+            case "h", "hr", "hrs", "hour", "hours" -> TimeUnit.HOURS;
+            case "d", "day", "days" -> TimeUnit.DAYS;
+            default -> throw new IllegalArgumentException("Could not parse '" + rawUnit + "' to TimeUnit");
+        };
+    }
+
+    private static String join(final String delimiter, final String... values) {
+        if (values.length == 0) {
+            return "";
+        } else if (values.length == 1) {
+            return values[0];
+        }
+
+        final StringBuilder sb = new StringBuilder();
+        sb.append(values[0]);
+        for (int i = 1; i < values.length; i++) {
+            sb.append(delimiter).append(values[i]);
+        }
+
+        return sb.toString();
+    }
```

Comment:
Does this method do something different than `String.join` ? If not I would suggest replacing the use of this with `String.join`.

Reply:
Yeah, good point. I pulled it over directly from FormatUtils, but I think now that we depend on Java 21 we can just use String.join